sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/TextAnalyticsAsyncClient.java

```diff
@@ -162,13 +161,23 @@ public TextAnalyticsServiceVersion getServiceVersion() {
      * @throws TextAnalyticsException if the response returned with an {@link TextAnalyticsError error}.
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
-    public Mono<Response<DetectedLanguage>> detectLanguageWithResponse(String text, String countryHint) {
-        try {
-            return withContext(context ->
-                detectLanguageAsyncClient.detectLanguageWithResponse(text, countryHint, context));
-        } catch (RuntimeException ex) {
-            return monoError(logger, ex);
-        }
+    public Mono<DetectedLanguage> detectLanguage(String text, String countryHint) {
+        return detectLanguageBatch(Collections.singletonList(text), countryHint, null)
+            .map(documentCollection -> {
+                final Iterator<DetectLanguageResult> iterator = documentCollection.iterator();
+                // Collection will never empty
+                if (!iterator.hasNext()) {
+                    throw logger.logExceptionAsError(
+                        new IllegalStateException("An empty collection returned which is an unexpected error."));
+                }
+
+                final DetectLanguageResult languageResult = iterator.next();
+                if (languageResult.isError()) {
+                    throw logger.logExceptionAsError(Transforms.toTextAnalyticsException(languageResult.getError()));
```

Comment:
You might want to raise the exception with the inner error here when the error is null.

Reply:
`documentCollection` 's error is either the `InnerError` or `TextAnalyticsError`. 

before calling .map(), the caller already take care of the scenario transform,  such that     `
```
Mono<Response<DocumentResultCollection<DetectLanguageResult>>> detectLanguageBatchWithResponse() 

called toDocumentResultCollection() which called Transforms.toTextAnalyticsError()
```